services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/liitteet/AttachmentsController.kt

```diff
@@ -0,0 +1,131 @@
+package fi.hel.haitaton.hanke.liitteet
+
+import fi.hel.haitaton.hanke.HankeError
+import fi.hel.haitaton.hanke.HankeNotFoundException
+import fi.hel.haitaton.hanke.HankeService
+import fi.hel.haitaton.hanke.currentUserId
+import fi.hel.haitaton.hanke.permissions.PermissionCode
+import fi.hel.haitaton.hanke.permissions.PermissionService
+import io.swagger.v3.oas.annotations.Operation
+import java.net.URLConnection
+import java.util.UUID
+import javax.servlet.http.HttpServletResponse
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.http.HttpEntity
+import org.springframework.http.HttpHeaders
+import org.springframework.http.HttpStatus
+import org.springframework.http.MediaType
+import org.springframework.web.bind.annotation.DeleteMapping
+import org.springframework.web.bind.annotation.ExceptionHandler
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.ResponseStatus
+import org.springframework.web.bind.annotation.RestController
+
+@RestController
+@RequestMapping("/liitteet")
+class AttachmentsController(
+    @Autowired private val attachmentService: AttachmentService,
+    @Autowired private val hankeService: HankeService,
+    @Autowired private val permissionService: PermissionService,
+) {
+
+    @GetMapping("/{uuid}")
+    @Operation(
+        summary = "Get attachment metadata by UUID",
+        description =
+            """
+                Return information about an attachment. Available even when scan is pending.
+            """
+    )
+    fun getAttachment(@PathVariable uuid: UUID): AttachmentMetadata {
+        val currentUserId = currentUserId()
+        val info = attachmentService.get(uuid)
+        val hankeId = hankeService.getHankeId(info.hankeTunnus)
+
+        if (
+            hankeId == null ||
+                !permissionService.hasPermission(hankeId, currentUserId, PermissionCode.VIEW)
+        ) {
+            throw HankeNotFoundException("")
```

Comment:
Shouldn't this throw `AttachmentNotFoundException` for the methods in this file, since there's no hanke in the URL?